remote.go

```diff
@@ -173,6 +173,37 @@ type Remote struct {
 	repo      *Repository
 }
 
+type RemoteDetached struct {
+	r Remote
+}
+
+// NewRemoteDetached returns a detached remote that has no
+// reference to a repository.  Used to emulate `git ls-remote <repo-url>`
+func NewRemoteDetached(remoteUri string) (*RemoteDetached, error) {
+	var ptr *C.git_remote
+
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	ret := C.git_remote_create_detached(&ptr, C.CString(remoteUri))
+	if ret != 0 {
+		return nil, MakeGitError(ret)
+	}
+
+	return &RemoteDetached{r: Remote{ptr: ptr}}, nil
```

Comment:
would it be possible for this to return the `*Remote` directly? That way the `RemoteDetached` and the `LsDetached` shouldn't be needed, since callers can interact with the remote using the normal API, which will avoid having to implement more APIs on `RemoteDetached` that would only forward arguments to the underlying `Remote`. It's totally fine if the `repo` property is `nil`: it's just there to avoid the GC from deleting the `Repository` when the `Remote` is still around, but here it will be `nil`.

also, the finalizer for the `Remote` object must be set to avoid memory leaks:

https://github.com/libgit2/git2go/blob/75708eea9844a3ae7cd48ace71d50fe8fbfc91c1/remote.go#L586